Content.Shared/Pulling/Systems/SharedPullerSystem.cs

```diff
@@ -15,6 +16,21 @@ public override void Initialize()
 
             SubscribeLocalEvent<SharedPullerComponent, PullStartedMessage>(PullerHandlePullStarted);
             SubscribeLocalEvent<SharedPullerComponent, PullStoppedMessage>(PullerHandlePullStopped);
+            SubscribeLocalEvent<SharedPullerComponent, VirtualItemDeletedEvent>(OnVirtualItemDeleted);
+        }
+
+        private void OnVirtualItemDeleted(EntityUid uid, SharedPullerComponent component, VirtualItemDeletedEvent args)
+        {
+            if (component.Pulling == null)
+                return;
+
+            if (component.Pulling == EntityManager.GetEntity(args.BlockingEntity));
```

Comment:
Does this throw if the entity itself is being deleted?

Reply:
`component.Pulling` and `args.BlockingEntity` aren't getting deleted, the virtual item entity in the hands is